cardano-cli/src/Cardano/CLI/Run/Mnemonic.hs

```diff
@@ -0,0 +1,184 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+
+module Cardano.CLI.Run.Mnemonic (generateMnemonicImpl, extendedSigningKeyFromMnemonicImpl) where
+
+import           Cardano.Api
+                   (AsType (AsCommitteeColdExtendedKey, AsCommitteeHotExtendedKey, AsDRepExtendedKey, AsPaymentExtendedKey, AsStakeExtendedKey),
+                   ExceptT, File, FileDirection (Out), HasTextEnvelope, Key (SigningKey),
+                   MnemonicSize (..), MnemonicToSigningKeyError, MonadIO (..), SerialiseAsBech32,
+                   except, findMnemonicWordsWithPrefix, firstExceptT, generateMnemonic, left,
+                   newExceptT, readTextFile, serialiseToBech32, signingKeyFromMnemonic,
+                   signingKeyFromMnemonicWithPaymentKeyIndex, textEnvelopeToJSON,
+                   writeLazyByteStringFile, writeTextFile)
+
+import qualified Cardano.CLI.Commands.Key as Cmd
+import           Cardano.CLI.Types.Common (KeyOutputFormat (..), SigningKeyFile)
+import           Cardano.CLI.Types.Errors.KeyCmdError
+                   (KeyCmdError (KeyCmdMnemonicError, KeyCmdReadMnemonicFileError, KeyCmdWriteFileError, KeyCmdWrongNumOfMnemonics))
+import           Cardano.Prelude (isSpace)
+
+import           Control.Monad (when)
+import           Data.Bifunctor (Bifunctor (..))
+import           Data.Text (Text)
+import qualified Data.Text as T
+import           Data.Word (Word32)
+import           System.Console.Haskeline (Completion, InputT, Settings (..), completeWord',
+                   defaultBehavior, defaultPrefs, getInputLineWithInitial,
+                   runInputTBehaviorWithPrefs, simpleCompletion)
+import           System.Console.Haskeline.Completion (CompletionFunc)
+
+-- | Generate a mnemonic and write it to a file or stdout.
+generateMnemonicImpl
```

Comment:
The `Impl` suffix is unnecessary. It would be similar to naming it `generateMnemonicFunction`.

Reply:
It was to avoid collision with the function from `Cardano.Api` with the same name. But I can just qualify import.

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/pull/975/commits/7c61f48a46114e70c0216846bd4a3c100d8a1a31